.github/integration_tests/nonexistent_repo.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Integration test to ensure CI fails when a non-existent repository is added
+
+# set git credentials
+git config user.name dfinity
+git config user.email dfinity@dfinity.org
+
+gh auth login
+
+# create new branch
+git checkout -b integration-test-1
+
+# add a fake repository
+echo "fake-repository" >> open-repositories.txt
+git add open-repositories.txt
+git commit -m 'add nonexistent repository'
+
+# create pull request
+git push --set-upstream origin integration-test-1
+gh pr create --draft --title "Integration Test 1" --body "generated by integration test"
+
+commit_sha=$(git rev-parse HEAD)
+
+# check CI
+while true
+do
+    sleep 3
+    status=$(curl -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_TOKEN" -H "X-GitHub-Api-Version: 2022-11-28"   https://api.github.com/repos/dfinity/repositories-open-to-contributions/actions/runs?branch=integration-test-1 | jq '[.workflow_runs[] | select(.name=="Check New Repo")] | map(select(.head_sha=='\"$commit_sha\"')) | max_by(.run_number) | .status')
+    echo $status
+    if [[ $status == '"completed"' ]]
+    then
+        break
+    fi
+done
+result=$(curl -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GH_TOKEN" -H "X-GitHub-Api-Version: 2022-11-28"   https://api.github.com/repos/dfinity/repositories-open-to-contributions/actions/runs?branch=integration-test-1 | jq '[.workflow_runs[] | select(.name=="Check New Repo")] | map(select(.head_sha=='\"$commit_sha\"')) | max_by(.run_number) | .conclusion')
```

Comment:
You could also use `gh pr checks`, but not sure it will look nicer.

Reply:
It does indeed look a little prettier, but it's not as flexible. I don't see a way to get a result of a specific workflow or filter by commit. I think working with the raw json gives me more options.

<img width="716" alt="Screenshot 2023-04-25 at 09 56 02" src="https://user-images.githubusercontent.com/47304080/234211870-3f29feee-b4e9-467a-ad2f-f95bfb34b5e3.png">